### PR TITLE
[feat]: [PIE-7529]: Merging input-sets version V1

### PR DESCRIPTION
**Pull request by brijeshdhakar**
## Describe your changes

Merging input-sets version V1.
Impact: there should be no impact. This flow id called only when Input-sets Version 1 are used.
Testing: Tested that input-set merge is happening properly for existing input-sets and V1 input-sets

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42404)

### CODE REVIEW

File: 884-pms-commons/src/test/java/io/harness/pms/merger/helpers/InputSetMergeHelperTest.java [hunk 1]
```diff
@@ -358,4 +359,28 @@ public void testMergeOnPipelineVMInfrastructure() {
...
+        + "name: partialset1\n"
+        + "inputs:\n"
+        + "  image: alpine\n"
+        + "  repo: harness-core\n"
+        + "  count: 0");
+
+    inputSetYamlList.add("version: 1\n"
+        + "name: partialset2\n"
+        + "inputs:\n"
+        + "  count: 1\n");
```
Comment (namanvermah) [inline 884-pms-commons/src/test/java/io/harness/pms/merger/helpers/InputSetMergeHelperTest.java:378]: Can you add a field that is only in partialset2 and not in partialset1 just to confirm that new fields are also present in the merged yaml?

**Review by brijeshdhakar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42404)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`884-pms-commons/src/test/java/io/harness/pms/merger/helpers/InputSetMergeHelperTest.java` line 378 at r2](https://reviewable.io/reviews/harness/harness-core/42404#-NLMvGLyJhNjJr_Soo5p-r2-378:-NLPJODS59dbx3LiVrSE:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d82b626efe3bbcb6649e2448e2803db879220ff6/884-pms-commons/src/test/java/io/harness/pms/merger/helpers/InputSetMergeHelperTest.java#L378)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

Can you add a field that is only in partialset2 and not in partialset1 just to confirm that new fields are also present in the merged yaml?
</blockquote></details>

Done.

File: 884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetMergeHelper.java [hunk 2]
```diff
@@ -63,6 +69,29 @@ public String mergeInputSetsForGivenStages(
...
+      } catch (IOException e) {
+        throw new InvalidRequestException(String.format("Input set is invalid: %s", inputSetYaml));
+      }
+      if (mergedInputSetNode == null) {
+        mergedInputSetNode = inputSetNode;
+      } else {
+        JsonNodeUtils.merge(mergedInputSetNode, inputSetNode);
+      }
+    }
+    return YamlPipelineUtils.writeYamlString(
```
Comment (raghavharness) [inline 884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetMergeHelper.java:90]: We should only add repository in response key if it's present in the final merge yaml

**Comment by brijeshdhakar** [comment]
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetMergeHelper.java` line 90 at r4](https://reviewable.io/reviews/harness/harness-core/42404#-NLMsfq2a60W1xM_i6qD-r4-90:-NLQ1ED719jhUrkhX4pX:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/11cadffd5f2bf8d16ced152b9743be7307269b72/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetMergeHelper.java#L90)):*
<details><summary><i>Previously, raghavharness (Raghav) wrote…</i></summary><blockquote>

We should only add repository in response key if it's present in the final merge yaml
</blockquote></details>

Done.

**Review by raghavharness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r2, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42404#-:-NLQ5s8p7Kvpqsr82Gw5:bnfp4nl)* status: 3 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by namanvermah — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42404)* status: 1. unresolved discussions.

**Review by raghavharness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42404)* status: :shipit: complete! ✓

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42404)* status: :shipit: complete! ✓

**Pull request merged**